book/src/spec.md

```diff
@@ -0,0 +1,628 @@
+Physical stream specification
+=============================
+
+A Tydi physical stream carries a stream of elements, dimensionality information
+for said elements, and (optionally) user-defined transfer information from a
+source to a sink. The contents of these three groups and to some extent the way
+in which they are transferred are defined in a parameterized way, to provide
+the necessary flexibility for the higher levels of this specification, and to
+give hardware developers the freedom to optimize their designs. Based on these
+parameters, this specification defines which signals must exist on the
+interface, and what the significance of these signals is.
+
+Parameters
+----------
+
+A physical stream interface has five parameters, described in the following
+subsections.
+
+> Note that we define the parameters on the interfaces of the source and the
+> sink, not on the stream between them! This is because the complexity
+> parameter need not be the same on the source and sink. The others do.
+
+### Element content (E)
```

Comment:
It's not necessary for this layer of the spec, so I simplified it away. Of course implementations are free to represent the list of elements with some recursive structure.

Reply:
> If we use the letter E for the total number of bits, we should probably not put it in the header here, since Element content is the emptyable, ordered set of named fields etc..

Fixed in 129bbb9c